prog/weaver/discovery.go

```diff
@@ -79,6 +89,12 @@ func HandleHTTPPeer(router *mux.Router, alloc *ipam.Allocator) {
 
 	router.Methods("DELETE").Path("/peer/{id}").HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		ident := mux.Vars(r)["id"]
+		if discoveryEndpoint != "" {
+			// TODO: deal with this being either a peername or a nickname
```

Comment:
*Minor*: ICYMI, you may want to deal with this `TODO` out before we merge the PR?